### DISCUSSION TITLE
splitting STL based on contours

### DISCUSSION THREAD

**Discussion by zebanaqviWAT** [pyvista/pyvista#4257] [Q&A] [answered]
Hi,
I have an STL file of a terrain surface, and need to slice it into different STL files generated by cutting along the contours, (possibly like contour_banded).
Ultimate goal is to be able to assign different boundary conditions to the bands.
Slice_along_lines seems to be a candidate that might help when I get to cutting stage.
But I am yet to figure how to use contour() on STL file.
Question 1 - How to generate contours on an STL file (read as pv.read(<STL file>))
Question 2 - How to cut it out along the contours and save different STL's for each region between neighboring contours.
Thanks!

**Comment by akaszynski** [comment, marked as answer]
Here's a follow up based on an offline conversation.

The question was to how to remove and split any geometry below a certain Z level and write them out to individual files.

From data obtained from [Moon2STL](https://jthatch.com/Moon2STL/), we have: [moon.zip](https://github.com/pyvista/pyvista/files/11224709/moon.zip)

Unzip this and run the following:

```py
import pyvista as pv
pv.set_plot_theme('document')

# read and remove the points at zero to make this the surface of just the moon.
mesh = pv.read('moon-1665.stl')
mesh, _ = mesh.remove_points(mesh.points[:, 2] == 0)

# Split anything above Z = 10 and below
below, above = mesh.clip(return_clipped=True, normal='z', origin=(0, 0, 10.0))

# next, split each individual body and add back in the original
split = below.split_bodies()
split.append(above)

# plot this
split.plot(multi_colors=True)

# you can now write to disk
# for ii in range(len(split)):
#     split[ii].extract_surface().save(f'seg_{ii}.stl')

```
![plot](https://user-images.githubusercontent.com/11981631/231858068-a889f2d2-7fdd-471f-8e1f-2eaa2a8920ba.png)
